xGame/Java/xgame/src/main/java/com/xgame/service/GameService.java

```diff
@@ -104,4 +107,28 @@ private String stringifyBoard() throws JsonProcessingException {
 		mapper.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
 		return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(this.board.getBoard());
 	}
+	
+	public List<String> getLegalMoves(int matchId, String piecePosition)
+			throws IllegalPositionException, JsonMappingException, JsonProcessingException {
+		ArrayList<String> legalMoves = new ArrayList<>();
+
+		this.resumeMatch(matchId);
+
+		ChessPiece piece;
+		try {
+			piece = board.getPiece(piecePosition);
+
+			if (piece == null) {
+				return legalMoves;
+			}
+
+			System.out.println(piece);
+			legalMoves = piece.legalMoves();
+
+		} catch (IllegalPositionException e1) {
+			throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
+					"Cannot get legal moves for " + piecePosition + ". Position is illegal.", e1);
+		}
```

Comment:
you're already catching this in the controller. Pick one or the other.

Reply:
Removed from controller